PowerShellGet/PSModule.psm1

```diff
@@ -1411,8 +1496,10 @@ function Find-Module
                 $isRepositoryNullOrPSGallerySpecified = $true
             }
         }
-		
-		PackageManagement\Find-Package @PSBoundParameters | Microsoft.PowerShell.Core\ForEach-Object {
+        
+        $results = PackageManagement\Find-Package @PSBoundParameters

```

Comment:
Any specific reason for changing this pipeline logic? It impacts the performance of first result.